corehq/apps/users/models.py

```diff
@@ -2050,6 +2055,12 @@ def __repr__(self):
             self=self
         ))
 
+    @skippable_quickcache(['self._id'], lambda _: settings.UNIT_TESTING)
+    def get_usercase_id(self):
+        from corehq.apps.hqcase.utils import get_case_by_domain_hq_user_id
+        usercase = get_case_by_domain_hq_user_id(self.domain, self._id, USERCASE_TYPE)
```

Comment:
The approach I would take is to make a new function (almost identical
to `get_case_by_domain_hq_user_id`)
which just returns the ID (`include_docs=False`). You should be able to
make a private function that both `get_case_by_domain_hq_user_id` and
`get_case_id_by_domain_hq_user_id`
can call.

On 21 August 2015 at 08:15, Norman Hooper <notifications@github.com> wrote:

> In corehq/apps/users/models.py
> <https://github.com/dimagi/commcare-hq/pull/7915#discussion_r37608830>:
>
> > @@ -2050,6 +2055,12 @@ def __repr__(self):
> >              self=self
> >          ))
> >
> > +    @skippable_quickcache(['self._id'], lambda _: settings.UNIT_TESTING)
> > +    def get_usercase_id(self):
> > +        from corehq.apps.hqcase.utils import get_case_by_domain_hq_user_id
> > +        usercase = get_case_by_domain_hq_user_id(self.domain, self._id, USERCASE_TYPE)
>
> You make a good point, @snopoke <https://github.com/snopoke>. I looked
> for a query to fetch just a case ID, and all I found was
> get_case_ids_in_domain_by_owner which also seems like overkill -- it
> would be nice to be able to specify a case type. Would it be better to use
> get_case_ids_in_domain_by_owner, or would we want to write something new
> that accepts a case type too?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/7915/files#r37608830>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa